uaclient/actions.py

```diff
@@ -161,6 +162,34 @@ def attach_with_token(
 
     machine_token_file.write(new_machine_token)
 
+    current_series = system.get_release_info().series
+    if current_series:
+        support_entitlement = (
+            machine_token_file.entitlements()
+            .get("support", {})
+            .get("entitlement")
+        )
+        if support_entitlement:
+            allowed_series = support_entitlement.get("affordances", {}).get(
+                "onlySeries"
+            )
+            if allowed_series and allowed_series != "none":
```

Comment:
I don't think allowed_series will ever be the string "none" (though it will usually be `None`)

We can combine some of these checks to reduce nesting

```suggestion
        only_series = (
            machine_token_file.entitlements()
            .get("support", {})
            .get("entitlement", {})
            .get("affordances", {})
            .get("onlySeries", None)
        )
```